bundles/org.openhab.binding.kostalinverter/src/main/java/org/openhab/binding/internal/kostal/inverter/secondgeneration/SecondGenerationHandler.java

```diff
@@ -0,0 +1,354 @@
+                valueConfiguration = command.toString();
+                dxsEntriesConf = "33556484";
+                preSetExecuteConfigurationChanges(httpClient, url, username, password, dxsEntriesConf,
+                        valueConfiguration);
+                break;
+            case SecondGenerationBindingConstants.CHANNEL_BATTERYCHARGETIMEFROM:
+                valueConfiguration = command.toString();
+                dxsEntriesConf = "33556239";
+                preSetExecuteConfigurationChanges(httpClient, url, username, password, dxsEntriesConf,
+                        valueConfiguration);
+                break;
+            case SecondGenerationBindingConstants.CHANNEL_BATTERYCHARGETIMETO:
+                valueConfiguration = command.toString();
+                dxsEntriesConf = "33556240";
+                preSetExecuteConfigurationChanges(httpClient, url, username, password, dxsEntriesConf,
+                        valueConfiguration);
+                break;
+            case SecondGenerationBindingConstants.CHANNEL_MAXDEPTHOFDISCHARGE:
+                valueConfiguration = command.toString();
+                dxsEntriesConf = "33556247";
+                preSetExecuteConfigurationChanges(httpClient, url, username, password, dxsEntriesConf,
+                        valueConfiguration);
+                break;
+            case SecondGenerationBindingConstants.CHANNEL_SHADOWMANAGEMENT:
+                valueConfiguration = command.toString();
+                dxsEntriesConf = "33556483";
+                preSetExecuteConfigurationChanges(httpClient, url, username, password, dxsEntriesConf,
+                        valueConfiguration);
+                break;
+            case SecondGenerationBindingConstants.CHANNEL_EXTERNALMODULECONTROL:
+                valueConfiguration = command.toString();
+                dxsEntriesConf = "33556482";
+                preSetExecuteConfigurationChanges(httpClient, url, username, password, dxsEntriesConf,
+                        valueConfiguration);
+                break;
+            case SecondGenerationBindingConstants.CHANNEL_INVERTERNAME:
+                valueConfiguration = command.toString();
+                dxsEntriesConf = "16777984";
+                preSetExecuteConfigurationChanges(httpClient, url, username, password, dxsEntriesConf,
+                        valueConfiguration);
+                break;
+        }
+    }
+
+    @Override
+    public void initialize() {
+        channelConfigs = SecondGenerationChannelConfiguration.getChannelConfiguration();
+        channelConfigsExt = SecondGenerationChannelConfiguration.getChannelConfigurationExt();
+        channelConfigsExtExt = SecondGenerationChannelConfiguration.getChannelConfigurationExtExt();
+
+        updateStatus(ThingStatus.UNKNOWN);
+
+        scheduler.scheduleWithFixedDelay(() -> {
+            try {
+                updateStatus(ThingStatus.ONLINE);
```

Comment:
Better set the Thing to ONLINE after the communication has succeeded. This avoids flickering of the state.